envs/monkey_zoo/terraform/firewalls.tf

```diff
@@ -96,5 +97,158 @@ resource "google_compute_firewall" "tunneling2-out" {
   }
 
   direction = "EGRESS"
-  destination_ranges = ["10.2.0.0/24"]
+  destination_ranges = ["10.2.0.0/27"]
+}
+
+resource "google_compute_firewall" "allow-tunneling-only-from-islands" {
+  name    = "${local.resource_prefix}allow-tunneling-only-from-islands"
+  network = google_compute_network.monkeyzoo.name
+
+  allow {
+    protocol = "all"
+  }
+
+  direction = "INGRESS"
+  priority = "1000"
+  source_tags = ["island"]
+  target_tags = ["tunneling-9"]
```

Comment:
Are these machines created with tags? Maybe just use the IP's instead

Reply:
Fixed by adding a tag, nice catch!